File: backend/Models/Concert.cs
using System;
using System.Collections.Generic;

namespace backend.Models;

public partial class Concert
{
    public int Id { get; set; }

    public DateTime? Date { get; set; }

    public string? Title { get; set; }

    public string? ConcertType { get; set; }

    public string? Venue { get; set; }

    public string? Source { get; set; }

    public string? DetailsLink { get; set; }

    public DateTime? ModifiedAt { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<Composer> Composers { get; set; } = new List<Composer>();

    public virtual ICollection<Programme> Programmes { get; set; } = new List<Programme>();
}

File: frontend/src/utils/Concert.ts
export type Concert = {
    date: Date;
    title: string;
    concertType: string;
    source: string;
    venue: string;
    programme: Array<Programme>;
    composers: Array<Composer>;
    detailsLink: string;
};

export type Composer = {
    composer1: string;
    [key: string]: any;
};

export type Programme = {
    piece: string;
    [key: string]: any;
};
